### PR TITLE
OpenCV3 compatibility

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Hi,
This pull request addresses the compatibility issues of pupil with OpenCV3. Most of the changes are taken from https://github.com/pupil-labs/pupil/tree/opencv3 (resolving the merge conflicts of this branch with `master`), but some additional fixes were added.

I am working with this for a while now, and it seems to be stable. However, I have not yet used all the features offered by pupil, so there might still be some issues left to fix.

Best, Tobi

### CODE REVIEW

File: pupil_src/player/vis_light_points.py [hunk 1]
```diff
@@ -46,15 +46,16 @@ def update(self,frame,events):
...
 
         # fix for opencv binding inconsitency
         if type(out)==tuple:
             out = out[0]
 
         overlay =  1/(out/falloff+1)
 
-        img *= cv2.cvtColor(overlay,cv2.COLOR_GRAY2RGB)
+        img_float = img.astype(np.float32) * cv2.cvtColor(overlay,cv2.COLOR_GRAY2RGB)
+        img.data = img_float.round().astype(np.uint8).data
```
Comment (Tobias-Fischer) [inline pupil_src/player/vis_light_points.py]: This is a dirty hack, but I couldn't get it to work in a nicer way. Does anyone have an idea on how to implement this in a nicer way?

**Comment by Tobias-Fischer** [comment]
By the way, I guess this breaks compatibility with OpenCV2, so it will require all users to change to OpenCV3 ..

**Comment by mkassner** [comment]
Thanks for the PR. We plan to merge this with our next release.

**Comment by willpatera** [comment]
@Tobias-Fischer - I merged your work, added a few more fixes for `OpenCV 3` compatibility, merged changes from master (`v0.8.5` release) so that the `opencv3` branch is now compatible with `macOS 10.12` in #486. 

Please note that I have removed `ROS` from the `include_dirs` in the cython `setup.py` files in the current `opencv3` branch.

**Comment by Tobias-Fischer** [comment]
Hi @willpatera, thanks! Can I ask you why you removed `ROS` from the `include_dirs`? As far as I know, additional directories don't harm, and using `OpenCV 3` provided by `ROS` is one of the easiest ways to use ROS on Ubuntu (which even in 16.04 doesn't provide `OpenCV 3` natively.
Best,
Tobias

**Comment by willpatera** [comment]
Hi @Tobias-Fischer - My apologies - I didn't realize that it was so difficult to use/build `OpenCV 3` on Ubuntu 16.04. We can add it back to the `include_dirs` along with other required `library_dirs` if you want to make a PR to the `opencv3` branch.

**Comment by Tobias-Fischer** [comment]
Hi @willpatera,
Yes, I think it was one of the worst decisions made by the Ubuntu team to stick with `OpenCV 2` for the long term support release ..
Yes, it would be great if we could add `ROS` back. I am busy these days, not sure when I'll get around to add it back.

Best,
Tobias

**Comment by Tobias-Fischer** [comment]
See https://github.com/pupil-labs/pupil/pull/488 instead

**Pull request closed without merging**